crates/ruff_linter/src/rules/pyflakes/rules/unused_import.rs

```diff
@@ -112,6 +114,36 @@ impl Violation for UnusedImport {
     }
 }
 
+fn is_first_party(checker: &Checker, qualified_name: &str) -> bool {
+    use isort::{ImportSection, ImportType};
+    let category = isort::categorize(
+        qualified_name,
+        0,
+        &checker.settings.src,
+        checker.package(),
+        checker.settings.isort.detect_same_package,
+        &checker.settings.isort.known_modules,
+        checker.settings.target_version,
+        checker.settings.isort.no_sections,
+        &checker.settings.isort.section_order,
+        &checker.settings.isort.default_section,
```

Comment:
Should we just pass default values for these?

Reply:
You mean `Default::default()` values? I don't know. I'm following the examples from the isort code that I saw elsewhere.

Reply:
Well, we're not sorting imports so I feel like it's weird to use these import sort section flags from the settings for categorization here. Maybe it's not... but I'd imagine setting values that make sense for this rule e.g. `no_sections` should be `false`, `section_order` should be an empty slice, and the `default_section` should be.. first party? so we don't remove things that we're not sure about? not sure.

Reply:
This function has `#[allow(clippy::too_many_arguments)]` on it and no docstring. I wasn't about to dig into its implementation to figure out the meaning of all the arguments, since the tests are passing and the behavior is correct. I'll do as you recommend and ensure the tests work. Thanks for looking.

Reply:
Oof, I tried those arguments and it starts miscategorizing stdin imports as first party. Hmm. More investigation is required.

Reply:
Ok, I tried various combinations of these arguments and got various incorrect categorization results, so I'm going to leave this as-is.

Reply:
I'm confused. It seems like using whatever our defaults are for the settings should give the correct behavior?

If a user changes these import sort settings, we shouldn't see a change of behavior in this rule — that sounds incorrect.

Reply:
We should use the user-defined settings IMO. I think what we have here is correct.

Reply:
I think we should use the user-settings for some parts, but why should changing your isort section order or default section affect this rule?

Reply:
I think including the default section _does_ make sense. From the perspective of this rule, these are opaque settings that the user provides to tell us how imports should be categorized. This rule shouldn't need to keep track of which settings _might_ matter based on details of its own logic. Imagine instead that this method just takes `settings.isort`. Re-creating that object to omit some values would be breaking the API contract. (And I don't really see any benefit -- it's actually _more_ work in the future, if we add arguments, to try and keep track of which ones should be provided and which should be intentionally omitted.)

Reply:
I mean.. the method doesn't take `settings.isort` but okay. It sounds like what I want is a different categorization method that doesn't rely on `isort` sections to determine if something is first-party or not — it's a weird mix of concerns. Perhaps I'm just misunderstanding the intent of these settings but, as a user, I would be very surprised that import-sorting options affect fixes for unused imports.